torch/_inductor/scheduler.py

```diff
@@ -861,6 +886,189 @@ def check_all_pairs(nodes):
 
         return sorted(possible_fusions, key=self.score_fusion_key, reverse=True)
 
+    @staticmethod
+    def _upper_triangle_pairs(nodes):
+        for node1_index, node1 in enumerate(nodes):
+            for node2 in nodes[node1_index + 1 :]:
+                yield node1, node2
+
+    @staticmethod
+    def reorder_deps(dep1, dep2) -> Optional[List[int]]:
+        if dep1 == dep2:
+            return None
+
+        if not isinstance(dep1, MemoryDep) or not isinstance(dep2, MemoryDep):
+            return None
```

Comment:
This is filtered below, so it could go. We could even type the arguments to this function.